### PR TITLE
Add COLLATE for MySQL

### PR DESCRIPTION
**Pull request by petertanath**

### CODE REVIEW

File: config/mysql.yml [hunk 1]
```diff
@@ -67,6 +67,7 @@ doctrine:
                  binary: DoctrineExtensions\Query\Mysql\Binary
                  char_length: DoctrineExtensions\Query\Mysql\CharLength
                  concat_ws: DoctrineExtensions\Query\Mysql\ConcatWs
+                 collate: DoctrineExtensions\Query\Mysql\Collate
```
Comment (stevelacey) [inline config/mysql.yml]: Up one line makes the list alphabetical ☝️

File: src/Query/Mysql/Collate.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        $parser->match(Lexer::T_CLOSE_PARENTHESIS);
+    }
+
+    /**
+     * @param \Doctrine\ORM\Query\SqlWalker $sqlWalker
+     * @return string
+     */
+    public function getSql(\Doctrine\ORM\Query\SqlWalker $sqlWalker)
+    {
+
```
Comment (stevelacey) [inline src/Query/Mysql/Collate.php]: Whitespace ✂️

File: src/Query/Mysql/Collate.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        return sprintf( '%s COLLATE %s', $sqlWalker->walkStringPrimary($this->stringPrimary), $this->collation );
+    }
+}
```
Comment (stevelacey) [inline src/Query/Mysql/Collate.php]: Add `\n`

File: tests/Query/Mysql/CollateTest.php [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    public function testCollate()
+    {
+        $q = $this->entityManager->createQuery("SELECT COLLATE('A',latin1_german2_ci) from DoctrineExtensions\Tests\Entities\Blank");
+
+        $this->assertEquals(
+            "SELECT 'A' COLLATE latin1_german2_ci AS sclr_0 FROM Blank b0_",
+            $q->getSql()
+        );
+    }
+
```
Comment (stevelacey) [inline tests/Query/Mysql/CollateTest.php]: ✂️

**Comment by stevelacey** [comment]
@petertanath thanks for your contribution :smile:

**Comment by petertanath** [comment]
Thanks! :)

**Pull request merged**